File: demos/jans-chip/ios/Jans-Chip/Models/Networking/AssertionOptionRequest.swift
//
//  AssertionOptionRequest.swift
//  Jans-Chip
//
//  Created by Nazar Yavornytskyi on 22.07.2024.
//

import Foundation

public struct AssertionOptionRequest {
    var username: String
    var userVerification: String?
    var documentDomain: String?
    var extensions: String?
    var session_id: String?
    var description: String?
}

File: demos/jans-chip/kmp/fido2/src/commonMain/kotlin/com/example/fido2/model/fido/assertion/option/AssertionOptionRequest.kt
package com.example.fido2.model.fido.assertion.option

import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json
import kotlinx.serialization.serializer

@Serializable
data class AssertionOptionRequest(
    var username: String?,
    var userVerification: String?,
    var documentDomain: String?,
    var extensions: String?,
    var session_id: String?,
    var description: String?
) {
    fun toJson() = Json.encodeToString(serializer(), this)
}